src/sql-parser/src/ast/defs/ddl.rs

```diff
@@ -405,6 +411,36 @@ impl<T: AstInfo> AstDisplay for Connector<T> {
 }
 impl_display_t!(Connector);
 
+/// Information about upstream Postgres tables used for replication sources
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+pub struct PgTable<T: AstInfo> {
+    /// The name of the table to sync
+    pub name: String,
```

Comment:
I think these should be changed to `UnresolvedObjectName`. We are parsing the column defs like a normal create table statement, I think we should parse the table name similarly.

Reply:
Just taking a look now, that makes sense to me! Fixing. Thanks @mjibson !

Reply:
You might have some difficulties with https://github.com/MaterializeInc/materialize/blob/c3259210a3cd966dbc44b07245ec035c786673e5/src/sql/src/postgres_util.rs#L40 but you can greatly simplify that whole query:

`format!("SELECT {}::regclass::oid", table)` (using params here didn't work in cockroach, but it might work in postgres. But the `blah::regclass::oid` thing is the magic that will get you the OID of the table you want).

Reply:
I hit the same problem when I tried to implement your suggestion in the other PR. Do you know if you can use parameters with this type of query? If not I'd personally stick with the original query

Reply:
@petrosagg would you want to switch back to avoid potential SQL injection? or something else?

Reply:
If we parse the name as a valid ObjectName then it should be ok, but if we accept it as a string literal I think we should use the long form query